### PR TITLE
Feat: Add next-themes for theme handling 

### LINKED ISSUE #3214: RFC: Use next-themes for theme handling
Right now we have a "semi-stable" theming engine. I propose delegating that responsibility to a battle-tested package like [next-themes](https://github.com/pacocoursey/next-themes).

### PR DESCRIPTION
**Pull request by G3root**
## What does this PR do?

Fixes #3214 

new dependency added:

- next-themes

**Environment**: Staging(main branch) / Production

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [ ] Test A
- [ ] Test B

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't performed a self-review of my own code and corrected any misspellings
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by PeerRich** [comment]
hey @G3root can you record a brief loom demo?

**Comment by G3root** [comment]
> hey @G3root can you record a brief loom demo?

https://loom.com/share/93e6773cade849a88ae04f73256542c0

**Comment by PeerRich** [comment]
left a comment in the loom at 0:55

is there a way to prevent the flicker?

**Comment by G3root** [comment]
> left a comment in the loom at 0:55
> 
> is there a way to prevent the flicker?

i will look onto it

File: apps/web/components/Shell.tsx [hunk 1]
```diff
@@ -664,4 +664,4 @@ function UserDropdown({ small }: { small?: boolean }) {
       </DropdownMenuContent>
     </Dropdown>
   );
-}
+}
```
Comment (zomars) [inline apps/web/components/Shell.tsx:667]: Small linting error here

File: apps/web/components/booking/CancelBooking.tsx [hunk 2]
```diff
@@ -32,90 +32,86 @@ export default function CancelBooking(props: Props) {
   const [loading, setLoading] = useState(false);
   const telemetry = useTelemetry();
   const [error, setError] = useState<string | null>(booking ? null : t("booking_already_cancelled"));
-  const { isReady, Theme } = useTheme(props.theme);
+  const {} = useTheme(props.theme);
```
Comment (zomars) [inline apps/web/components/booking/CancelBooking.tsx:35]: Is this still needed?

File: apps/web/pages/success.tsx [hunk 3]
```diff
@@ -249,337 +249,330 @@ export default function Success(props: SuccessProps) {
     return t("emailed_you_and_attendees" + titleSuffix);
   }
   const userIsOwner = !!(session?.user?.id && eventType.users.find((user) => (user.id = session.user.id)));
-  const { isReady, Theme } = useTheme(isSuccessBookingPage ? props.profile.theme : "light");
+  const {} = useTheme(isSuccessBookingPage ? props.profile.theme : "light");
```
Comment (zomars) [inline apps/web/pages/success.tsx:252]:
```suggestion
  useTheme(isSuccessBookingPage ? props.profile.theme : "light");
```

If we're not destructuring anything we can just do this

**Review by zomars — COMMENTED** [review]
Thank you for your contribution 🙏 Overall very nice work, will test locally and also tagging @hariombalhara to see if he has some thoughts.

**Review by zomars — APPROVED** [review]
Made some fixes and can confirm that is working even with custom branding. Nice work!

**Comment by PeerRich** [comment]
> Made some fixes and can confirm that is working even with custom branding. Nice work!

has the flicker (see loom comment) been fixed?

**Review by hariombalhara — DISMISSED** [review]
Theming doesn't seem to be working on booking pages.
https://www.loom.com/share/906782c770e24352b12447463452e6f4

I would be doing more testing in the meantime as it seems a tricky change to me. But blocking the PR from merging due to the known issue
I would be testing embeds as well.

**Comment by zomars** [comment]
Agreed, It should work as [their demo](https://next-themes-example.vercel.app)

**Review by hariombalhara — APPROVED** [review]
I fixed the bugs that I found. They are explained here https://www.loom.com/share/c9cfd3f4a0e94255ab460ee10c6a3397

This is good to go now.  Thanks, @G3root  for the contribution 🙏

**Pull request merged**